ext/drreg/drreg.c

```diff
@@ -441,6 +720,65 @@ drreg_event_bb_insert_late(void *drcontext, void *tag, instrlist_t *bb, instr_t
     }
 
     /* Before each app read, or at end of bb, restore spilled registers to app values: */
+#ifdef X86
+    for (reg = DR_REG_APPLICABLE_START_SIMD; reg <= DR_REG_APPLICABLE_STOP_SIMD; reg++) {
+        restored_for_simd_read[SIMD_IDX(reg)] = false;
+        if (!pt->simd_reg[SIMD_IDX(reg)].native) {
+            if (drmgr_is_last_instr(drcontext, inst) ||
+                instr_reads_from_reg(inst, reg, DR_QUERY_INCLUDE_ALL) ||
+                /* i#1954: for complex bbs we must restore before the next app instr */
```

Comment:
This works if the reserved register is xmm. But if it was ymm or zmm, a write to xmm would be like a partial write of GPRs. But then it is unclear whether it requires to restore a ymm or zmm?

And it needs even more scrutiny once drreg would consider AVX-512 masks. Then essentially any partial write to zmm could require a restore. We should add a comment about this here.

WDYT?

Reply:
This depends on what register has been reserved by the user. A user cannot reserve XMM0 and YMM0 separately. If the user reserves YMM0, and there is a partial read, the entire YMM0 is restored.

Reply:
What about a partial _write_? We restore partial _writes_ to gprs

Reply:
Sorry for missing your point on partial writes. If one reserves a YMM0 and a partial write to XMM0 is performed, then YMM0 needs to be restored just like gprs.